### DISCUSSION TITLE
Create documentation and examples for wagmi's internal caching

### DISCUSSION THREAD

**Discussion by MidnightLightning** [wevm/wagmi#3017] [Ideas]
Most `wagmi` hooks have `enabled` as a possible input parameter, and specify `refetch` return values that I think could be used together for use cases where the developer wants more control over when The data is updated, but I don't see documentation on them.

Additionally, I don't see documentation on what the difference is between `isLoading` and `isFetching`, and between `isFetched` and `isFetchedAfterMount` are supposed to represent.

Overall documentation on those common parameters would be helpful for developers to use to determine when/how `wagmi` internal caching would be appropriate for their app, or when/how to turn it off and bypass it in favor of their caching/refresh logic.

**Comment by tmm** [comment]
Good idea. Coming for Wagmi v2!

**Comment by MidnightLightning** [comment]
That doesn't help those of us currently using Wagmi v1... 😕 

"Just upgrade to the latest" isn't always a viable option for projects. Please consider reopening this issue, so the existing functionality can be well-documented for those who are using that version.

**Reply by tmm** [reply]
For the current Wagmi version, check out the [TanStack Query docs](https://tanstack.com/query/latest/docs/react/overview). Wagmi uses TanStack Query to manage async data.

**Reply by MidnightLightning** [reply]
Thanks; it's helpful to know that's the internal mechanism used. A documentation page on Wagmi pointing that fact out would be helpful. Additionally, even following the TanStack documentation, I'm unable to get the Wagmi return values to show different values for `isLoading`/`isFetching`; a documentation page with examples showing how those parameters are expected to fit into the React component lifecycle flow would be helpful.
